pkg/prometheus/resource_selector.go

```diff
@@ -121,6 +121,10 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 	for namespaceAndName, sm := range serviceMonitors {
 		var err error
 
+		if sm.Spec.ScrapeClass != nil {
```

Comment:
you'd to move this after the for loop otherwise the err might be swallowed.

Reply:
Can you review @simonpasquier ?